Make test module work in v2

msabramo opened the pull request:
Wasn't working because some utility functions have moved around:
- `jsonify` moved from `ansible.utils` to `ansible.parsing.utils.jsonify`
- I don't see `ansible.utils.parse_json` anymore so I used `json.loads`.

```
$ hacking/test-module -m lib/ansible/modules/core/packaging/os/apt_repository.py
* including generated source, if any, saving to: /Users/marca/.ansible_module_generated
* this may offset any line numbers in tracebacks/debuggers!
***********************************
RAW OUTPUT
{"msg": "missing required arguments: repo", "failed": true}


***********************************
PARSED OUTPUT
{
    "failed": true,
    "msg": "missing required arguments: repo"
}
```

sivel commented:
That file isn't missing, it has been moved to `ansible.executor.module_common`

Also if you are going to switch to using the print function you should also copy in the `__future__` imports that all files now have in v2

Commit:
Make test-module work in v2

- `jsonify` moved from `ansible.utils` to `ansible.parsing.utils.jsonify`
- I don't see `ansible.utils.parse_json` anymore so I used `json.loads`.

msabramo commented:
OK, revised. It's simpler now because the file isn't missing and I don't try to change the `print` statements to keep the PR smaller and more focused.

msabramo commented:
Hmmm why doesn't this work?

```
vagrant@localmonkey:~/dev/ansible$ hacking/test-module -m lib/ansible/modules/core/packaging/os/apt_repository.py -a repo=\"ppa:fkrull/deadsnakes\"
* including generated source, if any, saving to: /home/vagrant/.ansible_module_generated
* this may offset any line numbers in tracebacks/debuggers!
***********************************
RAW OUTPUT
{"msg": "missing required arguments: repo", "failed": true}


***********************************
PARSED OUTPUT
{
    "failed": true,
    "msg": "missing required arguments: repo"
}
```

I'm trying to follow along with what http://docs.ansible.com/developing_modules.html#reading-input shows.

msabramo commented:
Hmmm.

``` pytb
vagrant@localmonkey:~/dev/ansible$ hacking/test-module -m lib/ansible/modules/core/packaging/os/apt_repository.py -a repo=\"ppa:fkrull/deadsnakes\"
...
> /home/vagrant/dev/ansible/hacking/test-module(142)runtest()
-> invoke = "%s" % (modfile)
(Pdb) l
137     def runtest( modfile, argspath):
138         """Test run a module, piping it's output for reporting."""
139
140         os.system("chmod +x %s" % modfile)
141
142  ->     invoke = "%s" % (modfile)
143         if argspath is not None:
144             invoke = "%s %s" % (modfile, argspath)
145
146         cmd = subprocess.Popen(invoke, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
147         (out, err) = cmd.communicate()
(Pdb) argspath
(Pdb) p argspath
None
```

Looks like there is some new module style where the arguments are passed differently? And maybe that isn't quite working yet?

https://github.com/ansible/ansible/blob/devel/hacking/test-module#L179

``` pytb
> /home/vagrant/dev/ansible/hacking/test-module(188)main()
-> if options.debugger:
(Pdb) module_style
'new'
(Pdb) l
183                 argspath = write_argsfile(options.module_args, json=True)
184             elif module_style == 'old':
185                 argspath = write_argsfile(options.module_args, json=False)
186             else:
187                 raise Exception("internal error, unexpected module style: %s" % module_style)
188  ->     if options.debugger:
189             rundebug(options.debugger, modfile, argspath)
190         else:
191             runtest(modfile, argspath)
192
193     if __name__ == "__main__":
(Pdb) modfile
'/home/vagrant/.ansible_module_generated'
(Pdb) p argspath
None
```

Commit:
hacking/test-module: Deal with move of parse_kv

Commit:
hacking/test-module: Style nit
